hypothesis-python/src/hypothesis/core.py

```diff
@@ -1118,9 +1121,10 @@ def template_condition(data):
         with BuildContext(data):
             try:
                 data.is_find = True
-                result = data.draw(search)
-                data.note(result)
-                success = condition(result)
+                with deterministic_PRNG():
+                    result = data.draw(search)
+                    data.note(result)
+                    success = condition(result)
```

Comment:
If I’ve understood correctly, `data.draw(search)` is the only bit of code that actually draws any random data?

I’d prefer to close the context manager ASAP, and move lines 2 and 3 out of the block.

Reply:
No, the the entire test should run inside the context manager - it's to stop tests that the fail flakily in a way that depends on the global random state, then reset the random state after the test has finished.

Reply:
Okay, yeah, I think I see why the block can’t be tightened now.